File: solutions/0690. Employee Importance/0690.py
class Solution:
    def getImportance(self, employees: List['Employee'], id: int) -> int:
        def dfs(id: int) -> int:
            subordinatesImportance = sum(
                dfs(subordinate) for subordinate in idToEmployee[id].subordinates)
            return idToEmployee[id].importance + subordinatesImportance

        idToEmployee = {employee.id: employee for employee in employees}

        return dfs(id)

File: solutions/0690. Employee Importance/0690.cpp
class Solution {
 public:
  int getImportance(vector<Employee*> employees, int id) {
    unordered_map<int, Employee*> idToEmployee;

    for (Employee* employee : employees)
      idToEmployee[employee->id] = employee;

    function<int(int)> dfs = [&](int id) {
      int subordinatesImportance = idToEmployee[id]->importance;

      for (const int subordinate : idToEmployee[id]->subordinates)
        subordinatesImportance += dfs(subordinate);

      return subordinatesImportance;
    };

    return dfs(id);
  }
};